specs/WebMessageObjects.md

```diff
@@ -227,148 +463,32 @@ interface WebView extends EventTarget {
-            // parameter parsed as a JSON string into a JavaScript object. The event arg's
-            // <c>source</c> property of the event arg is a reference
-            // to the <c>window.chrome.webview</c> object. The <c>additionalObjects</c> property of
-            // event arg will contain a sequence of WebMessage objects as their DOM types. For
-            // information about sending messages from the HTML document in the WebView to the
-            // host, navigate to <see cref="CoreWebView2.WebMessageReceived"/>. The message is sent
-            // asynchronously. If a navigation occurs before the message is posted to the page,
-            // the message is not be sent.
-            // </remarks>
-            // <example>
-            // Runs the message event of the <c>window.chrome.webview</c> of the top-level
-            // document. JavaScript in that document may subscribe and unsubscribe to the event
-            // using the following code:
-            // $net$
-            // <code>
-            // window.chrome.webview.addEventListener('message', handler)
-            // window.chrome.webview.removeEventListener('message', handler)
-            // </code>
-            // $end$
-            // $winrt$
-            // ```javascript
-            // window.chrome.webview.addEventListener('message', handler)
-            // window.chrome.webview.removeEventListener('message', handler)
-            // ```
-            // $end$
-            // </example>
-            // <seealso cref="CoreWebView2Settings.IsWebMessageEnabled"/>
-            // <seealso cref="WebMessageReceived"/>
-            // <seealso cref="PostWebMessageAsString"/>
-            void PostWebMessageAsJsonWithAdditionalObjects(
-                String webMessageAsJson,
-                IVectorView<Object> additionalObjects);
-        }
-    }
-}
-```
+interface WebViewMessageEvent extends MessageEvent {
+    /**
+     * The source of the event is the `chrome.webview` object.
+     */
+    source: WebView;
 
-```ts
-interface WebViewMessageEvent: MessageEvent {
-  /* When a WebMessage is posted via PostWebMessageAsJsonWithAdditionalObjects
-   * this property will contain the DOM objects that are injected to the content.
-   * Currently the following types are supported:
-   * - File
-   */
-  additionalObjects: ArrayLike<any>
+    /**
+     * Additional DOM objects that are sent via PostJSONMessageWithAdditionalObjects.
+     * These objects can be of following types:
+     * DOM      | Win32
+     * -------- | ------------
+     * [File](https://developer.mozilla.org/docs/Web/API/File] | ICoreWebView2File
+     * [FileSystemFileHandle](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle) | ICoreWebView2FileSystemHandle (kind File)
+     * [FileSystemDirectoryHandle](https://developer.mozilla.org/docs/Web/API/FileSystemDirectoryHandle) ICoreWebView2FileSystemHandle (kind Directory)

```

Comment:
Not following the table syntax for this line and the next. Need to add the |. And for nullptr you could do like `undefined/null | nullptr`